src/openforms/formio/service.py

```diff
@@ -114,19 +123,61 @@ def build_serializer(
 
 def as_json_schema(
     component: Component, _register: ComponentRegistry | None = None
-) -> JSONObject:
+) -> JSONObject | list[JSONObject] | None:
     """Return a JSON schema of a component.
 
     A description will be added if it is available.
 
+    Layout components require some extra attention:
+      - Content and softRequiredErrors components do not have any values, so a
+        schema does not exist: returns ``None``
+      - Columns and fieldset components contain other components for which a JSON schema
+        needs to be generated: returns a list of JSON objects with the child component
+        key as key and the child component JSON schema as value.
+
     :param component: Component
-    :param _register: ComponentRegistry | None
-    :returns: JSONObject
+    :param _register: Optional component registry to use. If no registry was provided,
+      the default registry will be used.
+    :returns: None for content and softRequiredErrors components, list of JSON objects
+      for columns and fieldsets, and a JSON object otherwise.
     """
     registry = _register or register
 
-    component_plugin = registry[component["type"]]
-    schema = component_plugin.as_json_schema(component)
-    if description := component.get("description"):
-        schema["description"] = description
-    return schema
+    match component["type"]:
+        case "content" | "softRequiredErrors":
+            return None
+        case "columns":
+            component = cast(ColumnsComponent, component)
```

Comment:
Is `cast` added because of type checking complaints? Because I think we should generally avoid using `cast`.